dhcpv6/duid_test.go

```diff
@@ -104,3 +104,17 @@ func TestDuidUuidToBytes(t *testing.T) {
 		t.Fatalf("Invalid ToBytes result. Expected %v, got %v", expected, toBytes)
 	}
 }
+
+func TestOpaqueDuid(t *testing.T) {
+	duid := []byte("\x00\x0a\x00\x03\x00\x01\x4c\x5e\x0c\x43\xbf\x39")
+	d, err := DuidFromBytes(duid)
+	if err != nil {
```

Comment:
We are now using `require` for unit tests, and gradually migrating to it. Can you use it instead of manual checks?

E.g. this would become
```
require.NoError(t, err)
require.Equal(t, len(duid), d.Length())
require.Equal(t, duid, d.ToBytes())
```

Thanks!

Reply:
Sorry, no. Assert libraries are not idiomatic Go:

* they obscure control flow which is otherwise easily apparent
* they stop tests early (tests should keep going to require as few runs as possible)
* they encourage bad error messages

If you’re dead set on using an assert library regardless, please accept my commit as-is and change the test afterwards.

Thanks for understanding,

Reply:
No strong opinions on this, I will merge your PR as it is. Would you recommend a reading with more insight so we can discuss switching back away from assert libraries?

Reply:
https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures plays into it, but isn’t exactly focused on assert libraries. I’ll ask around.

Reply:
I’m not aware of a public document on this topic, aside from https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures, which I mentioned in my previous comment, sorry. Maybe it’s time to start one :)

Reply:
it would be a nice idea. It will be easier to push people to avoid common mistakes. Thanks!

Reply:
oh, if you do please let me know here or over email. Thanks!